src/App.tsx

```diff
@@ -1,223 +1,235 @@
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React, { useContext, useEffect, useRef } from 'react';
-import { AuthContext } from './components/Auth/AuthContext';
+import React, {
+  useContext, useEffect, useState,
+} from 'react';
+import {
+  getTodos, addTodo, deleteTodo, completeTodo, editTodo,
+} from './api/todos';
+import { AuthContext } from './components/Auth';
+import { Footer } from './components/Footer';
+import { TodosList } from './components/TodosList';
+import { ErrorNotification } from './components/ErrorNotification';
+import { Header } from './components/Header';
+import { Todo } from './types/Todo';
+import { Filter } from './types/Filter';
+import { ErrorType } from './types/ErrorType';
 
 export const App: React.FC = () => {
-  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [title, setTitle] = useState<string>('');
+  const [filter, setFilter] = useState(Filter.all);
+  const [error, setError] = useState<string>('');
+  const [hidden, setHidden] = useState(true);
+  const [adding, setAdding] = useState(false);
+  const [selectedTodoIds, setSelectedTodoIds] = useState<number[]>([]);
+  const [editing, setEditing] = useState(false);
+
   const user = useContext(AuthContext);
-  const newTodoField = useRef<HTMLInputElement>(null);
+  const completedTodos = todos.filter(todo => todo.completed);
 
   useEffect(() => {
-    // focus the element with `ref={newTodoField}`
-    if (newTodoField.current) {
-      newTodoField.current.focus();
+    if (user) {
+      getTodos(user.id)
+        .then(setTodos);
     }
   }, []);
 
+  const handleAdd = () => {
+    if (!user) {
+      return;
+    }
+
+    if (!title.trim()) {
+      setError('empty');
+      setHidden(false);
+      setTimeout(() => setHidden(true), 3000);
+
+      return;
+    }
+
+    setAdding(true);
+
+    const todo = {
+      title,
+      userId: user.id,
+      completed: false,
+    };
+
+    addTodo(todo)
+      .then(result => setTodos(
+        prevTodos => [...prevTodos, result],
+      ))
+      .catch(() => {
+        setError(ErrorType.add);
+        setHidden(false);
+      })
+      .finally(() => {
+        setAdding(false);
+        setTitle('');
+        setTimeout(() => setHidden(true), 3000);
```

Comment:
should we set this timer even if the is no error?

Reply:
think about moving error handling logic to a separate function to avoid repeating